src/main/java/io/Parser.java

```diff
@@ -0,0 +1,175 @@
+package io;
+
+import exceptions.ParserException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+
+/**
+ * The Parser class handles user input from the command line.
+ */
+public class Parser {
+
+    public Parser() {
+        inputTokens = null;
+        inputString = null;
+        scanner = new Scanner(System.in);
+    }
+
+    /**
+     * Waits for new input from the command line and updates internal variables.
+     *
+     * @throws NoSuchElementException If no further input is available.
+     */
+    public void update() throws NoSuchElementException {
+        inputString = scanner.nextLine();
+        inputTokens = inputString.split(" ");
+    }
+
+    /**
+     * updates the internal input string function
```

Comment:
Just a nit here, but perhaps you could capitalize the first word of your Javadoc comment and end the comment with a full stop?